spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -383,14 +390,52 @@ public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
-    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled", "true"));
-    Catalog catalog = buildIcebergCatalog(name, options);
+    this.cacheEnabled = PropertyUtil.propertyAsBoolean(options,
+        CatalogProperties.TABLE_CACHE_ENABLED, CatalogProperties.TABLE_CACHE_ENABLED_DEFAULT);
+
+    // If the user disabled caching and did not set the cache.expiration-interval-ms, we'll set it to zero for
+    // them on their behalf. If they disabled caching but explicitly set a non-zero cache expiration
+    // interval, we will fail initialization as that's an invalid configuration.
```

Comment:
I'm not sure I agree with this. I think that an `enabled` flag is compatible with other configuration. It is one simple config you can use to flip on or off a feature, without altering other properties. For example, if I were trying to debug something and I suspected caching, I'd go set this to disabled. It would annoy me at that point if Iceberg complained that I have caching disabled, but set a config property for it. So I'd probably override with `cache-enabled` and ignore that the interval is set or not.

Reply:
So we want two config properties now? I'm happy to add back an `enabled` flag if desired.

Just want to make sure as I had that before and it was requested to be removed. Happy to add it back in though as it makes sense to me.

Reply:
Updated.